components/omega/src/base/Halo.h

```diff
@@ -265,52 +317,66 @@ class Halo {
          TotSize *= Array.extent(NDims - 1);
       }
 
+      // Synchronize all tasks before starting communication
+      MPI_Barrier(MyComm);
```

Comment:
Is this barrier needed?  Ok if it is, just don't want to introduce any unnecessary sync points.

Reply:
I can remove it. I do kinda worry about race conditions that might occur with sequential exchanges one right after another, since the same buffers are used for each exchange, but nothing specific has popped up in tests so far. Can hold off on any extra sync points for now.

Reply:
Thanks - I understand the worry but if we have the proper barriers/waits after the messages and the subroutine exit ensures the buffer has been copied into the proper place, we should be ok.

Reply:
Slight clarification of that - we'll need to watch out for cases where we need kokkos fences too since a parallelFor is asynchronous.